File: doc-examples/example-kotlin/src/main/kotlin/io/micronaut/gcp/pubsub/client/MultipleProjectClient.kt
package io.micronaut.gcp.pubsub.client

//tag imports[]
import io.micronaut.gcp.pubsub.annotation.PubSubClient
import io.micronaut.gcp.pubsub.annotation.Topic
import io.micronaut.gcp.pubsub.support.Animal
// end::imports[]

// tag::clazz[]
@PubSubClient
interface MultipleProjectClient {

	@Topic("animals")
	fun sendUS(animal: Animal) // <1>

	@Topic("projects/eu-project/topics/animals")
	fun sendEU(animal: Animal) // <2>

}
// end::clazz[]

File: test-suite-python/src/main/python/micronaut/gcp/pubsub/client/MultipleProjectClient.py
from abc import ABC, abstractmethod

# tag::imports[]
from micronaut.gcp.pubsub.annotation import PubSubClient, Topic
from micronaut.gcp.pubsub.support.Animal import Animal
# end::imports[]


# tag::clazz[]
@PubSubClient
class MultipleProjectClient(ABC):

    @Topic("animals")
    @abstractmethod
    def send_us(self, animal: Animal) -> None:  # <1>
        ...

    @Topic("projects/eu-project/topics/animals")
    @abstractmethod
    def send_eu(self, animal: Animal) -> None:  # <2>
        ...
# end::clazz[]
